mitodl/mitxonline
# Implemented user-supplied usernames

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #191 

#### What's this PR do?
Requires users to specify a username during account creation

#### How should this be manually tested?
- Create a new account and follow the link to complete registration
- Try entering a username with special characters (&*^ etc...). You should see an error message after submitting.
- Try entering a username that obeys the validation rules. Your new user should be created with the custom username

#### Any background context you want to provide?
- The set of valid username characters comes from edX (https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L1346-L1347)
- Due to some shortcomings with Unicode in JS, it's very much not straightforward to implement front-end validation for this username field in a way that will allow for non-Latin letters, but will not allow special characters (&*^ etc...). For this first implementation, validation is just being done on the back-end

#### Screenshots (if appropriate)
![ss 2021-09-17 at 08 10 20 ](https://user-images.githubusercontent.com/14932219/133787897-23f2f3da-eded-48b7-aafb-093a577a80f6.png)
![ss 2021-09-17 at 08 32 29 ](https://user-images.githubusercontent.com/14932219/133791138-194a8ebe-71b0-4a15-a7d3-c19f0eda14ce.png)
![ss 2021-09-17 at 08 32 14 ](https://user-images.githubusercontent.com/14932219/133791143-53107b6b-7515-4f7c-84b7-bcbfe86279c4.png)
![ss 2021-09-17 at 13 03 54 ](https://user-images.githubusercontent.com/14932219/133833758-0e3e64a6-fc47-4bfe-9d24-838940d69d21.png)
